projects/aca-content/src/lib/services/content-management.service.ts

```diff
@@ -657,29 +657,60 @@ export class ContentManagementService {
 
   deleteNodes(items: NodeEntry[]): void {
     const batch: Observable<DeletedNodeInfo>[] = [];
+    const areHolds = items.some((node) => node.entry.nodeType === 'rma:hold');
+    const action = areHolds ? this.showWarningOnHoldDeleteAction(items) : of(true);
 
-    items.forEach((node) => {
-      batch.push(this.deleteNode(node));
-    });
+    action.pipe(take(1)).subscribe((removeApproval) => {
+      if (removeApproval) {
+        items.forEach((node) => {
+          batch.push(this.deleteNode(node));
+        });
 
-    forkJoin(...batch).subscribe((data: DeletedNodeInfo[]) => {
-      const status = this.processStatus(data);
-      const message = this.getDeleteMessage(status);
+        forkJoin(...batch).subscribe((data: DeletedNodeInfo[]) => {
+          const status = this.processStatus(data);
+          const message = this.getDeleteMessage(status);
 
-      if (message && status.someSucceeded) {
-        message.userAction = new SnackbarUserAction('APP.ACTIONS.UNDO', new UndoDeleteNodesAction([...status.success]));
-      }
+          if (message && status.someSucceeded) {
+            message.userAction = new SnackbarUserAction('APP.ACTIONS.UNDO', new UndoDeleteNodesAction([...status.success]));
+          }
 
-      this.store.dispatch(message);
+          this.store.dispatch(message);
 
-      if (status.someSucceeded) {
-        this.appHookService.nodesDeleted.next();
+          if (status.someSucceeded) {
+            this.appHookService.nodesDeleted.next();
+            this.documentListService.reload();
+          }
+        });
+      } else {
         this.documentListService.reload();
       }
       this.store.dispatch(new ShowLoaderAction(false));
     });
   }
 
+  showWarningOnHoldDeleteAction(nodes: NodeEntry[]): Observable<boolean> {
```

Comment:
You're leaking the enterprise code to community edition.. ACA should know nothing about legal hold, this should only be a part of governance

Reply:
I need to agree with @MichalKinas here. Please provide other solution.